dotnet/samples/04.e.twentyQuestions/Program.cs

```diff
@@ -72,14 +72,21 @@
         moderator: moderator,
         prompt: "Chat",
         history: aiHistoryOptions);
-    ApplicationOptions<GameState, GameStateManager> applicationOptions = new()
+    var applicationBuilder = new ApplicationBuilder<GameState, GameStateManager>()
+    .WithAIOptions(aiOptions)
+    .WithLoggerFactory(loggerFactory)
+    .WithTurnStateManager(new GameStateManager());
+
+    // Set storage options
+    IStorage? storage = sp.GetService<IStorage>();
+    if (storage != null)
     {
-        TurnStateManager = new GameStateManager(),
-        Storage = sp.GetService<IStorage>(),
-        AI = aiOptions,
-        LoggerFactory = loggerFactory,
-    };
-    Application<GameState, GameStateManager> app = new(applicationOptions);
+        applicationBuilder.WithStorage(storage);
+    }
+
+    // Create Application
+    Application<GameState, GameStateManager> app = applicationBuilder.Build();
```

Comment:
Instead of declaring the `applicationBuilder` and then doing `app = applicationBuilder.build()` you can do `app = ApplicationBuilder.withX().withY().Build()`.

I find that to be more concise. Thoughts?

Reply:
Good point! I separated the chaining because the `storage` object might be null (unless you have another idea to workaround) - same applies for `TwentyQuestions`

Reply:
I didn't notice that, this makes sense then. Thanks for clarifying.